arc/analysis/assumption_detector.py

```
@@ -0,0 +1,403 @@
+        """Fast pattern-based assumption detection."""
+        violations = []
+        
+        # Get trajectory text for analysis
+        trajectory_text = self._extract_trajectory_text(trajectory)
+        scenario_context = trajectory.get("scenario", {})
+        
+        # Check relevant patterns for each domain
+        for domain in domains + ["general"]:
+            domain_patterns = self.domain_patterns.get(domain, {})
+            
+            for pattern_name, pattern in domain_patterns.items():
```

Ruff (0.11.9)

Loop control variable `pattern_name` not used within loop body

Rename unused `pattern_name` to `_pattern_name`

(B007)